fenix/app/src/test/java/org/mozilla/fenix/gleanplumb/state/MessagingMiddlewareTest.kt

```diff
@@ -178,27 +195,36 @@ class MessagingMiddlewareTest {
     }
 
     @Test
-    fun `WHEN onMessageDismissed THEN updateMetadata,removeMessage , UpdateMessages and removeMessageToShowIfNeeded`() = runTestOnMain {
+    fun `WHEN onMessageDismissed THEN updateMetadata,removeMessage , UpdateMessages and consumeMessageToShowIfNeeded`() = runTestOnMain {
         val message = Message(
             "control-id",
             mockk(relaxed = true),
             action = "action",
             mockk(relaxed = true),
             listOf("trigger"),
-            Message.Metadata("same-id"),
+            Message.Metadata("same-id", displayCount = 0),
```

Comment:
For `onMessageDismissed` to happen, wouldn't we need to have displayed the message at least once?

```suggestion
            Message.Metadata("same-id", displayCount = 1),
```